### ISSUE TITLE
Accessibility: Modal cannot be closed by keyboard only

### ISSUE THREAD

**Issue by MichaelT372** [4site-interactive-studios/engrid-scripts#415]
[#modal](https://github.com/4site-interactive-studios/engrid/blob/main/packages/scripts/src/modal.ts)

This base modal class cannot be closed by keyboard only if the close button is not added (and we only have the X in top right corner). Add functionality to handle closing with escape key or keyboard event on the X.
